File: legacy/at_smartrak_gtfs_adapter/src/model/dead-reckoning.ts
import { transit_realtime } from "at-realtime-common/gtfs-realtime";

import ITripDescriptor = transit_realtime.ITripDescriptor;
export class DeadReckoningMessage {
    public id: string;
    public receivedAt: number;
    public position: PositionDR;
    public trip: ITripDescriptor;
    public vehicle: VehicleDR;

    constructor(id: string, receivedAt: number, position: PositionDR, trip: ITripDescriptor, vehicle: VehicleDR) {
        this.id = id;
        this.receivedAt = receivedAt;
        this.position = position;
        this.trip = trip;
        this.vehicle = vehicle;
    }
}
export class PositionDR {
    public odometer: number;

    constructor(odometer: number) {
        this.odometer = odometer;
    }
}
export class VehicleDR {
    public id: string;

    constructor(id: string) {
        this.id = id;
    }
}

File: crates/smartrak-gtfs/src/model/dead_reckoning.rs
use serde::{Deserialize, Serialize};

use super::trip::TripDescriptor;

// Mirrors message payload from legacy/at_smartrak_gtfs_adapter/src/model/dead-reckoning.ts.
#[derive(Debug, Clone, Serialize, Deserialize, Default)]
#[serde(rename_all = "camelCase")]
pub struct DeadReckoningMessage {
    pub id: String,
    pub received_at: i64,
    pub position: PositionDr,
    pub trip: TripDescriptor,
    pub vehicle: VehicleDr,
}

#[derive(Debug, Clone, Serialize, Deserialize, Default)]
#[serde(rename_all = "camelCase")]
pub struct PositionDr {
    pub odometer: f64,
}

#[derive(Debug, Clone, Serialize, Deserialize, Default)]
#[serde(rename_all = "camelCase")]
pub struct VehicleDr {
    pub id: String,
}
